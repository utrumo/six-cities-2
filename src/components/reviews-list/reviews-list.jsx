import React from 'react';
import ReviewsItem from '../reviews-item/reviews-item.jsx';
import PropTypes from 'prop-types';

import {connect} from 'react-redux';
import {getCommentsForOfferPage} from '../../store/data/selectors.js';

const ReviewsList = ({reviews}) => (
  <ul className="reviews__list">
    {reviews.map(({id, user, rating, comment, date}) => (
      <ReviewsItem
        key={id}
        user={user}
        rating={rating}
        comment={comment}
        date={date}
      />
    ))}
  </ul>
);

ReviewsList.propTypes = {
  reviews: PropTypes.arrayOf(PropTypes.exact({
    id: PropTypes.number.isRequired,
    user: PropTypes.exact({
      id: PropTypes.number.isRequired,
      isPro: PropTypes.bool.isRequired,
      name: PropTypes.string.isRequired,
      avatarUrl: PropTypes.string.isRequired,
    }).isRequired,
    rating: PropTypes.number.isRequired,
    comment: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
  })).isRequired,
};

const mapStateToProps = (state) => ({
  reviews: getCommentsForOfferPage(state),
});

export {ReviewsList};
export default connect(mapStateToProps)(ReviewsList);
