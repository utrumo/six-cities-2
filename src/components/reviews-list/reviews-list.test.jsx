import React from 'react';
import renderer from 'react-test-renderer';
import {ReviewsList} from './reviews-list.jsx';

const reviews = [
  {
    id: 1,
    user: {
      id: 17,
      isPro: false,
      name: `Emely`,
      avatarUrl: `/img/avatar-angelina.jpg`,
    },
    rating: 4,
    comment: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!`,
    date: `2019-11-20T13:49:50.988Z`,
  },
];

it(`ReviewsList component renders correctly after restart`, () => {
  const tree = renderer
    .create(<ReviewsList reviews={reviews} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
