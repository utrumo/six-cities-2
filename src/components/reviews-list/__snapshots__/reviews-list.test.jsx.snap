// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsList component renders correctly after restart 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="/img/avatar-angelina.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Emely
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!
      </p>
      <time
        className="reviews__time"
        dateTime="2019-11-20"
      >
        November 2019
      </time>
    </div>
  </li>
</ul>
`;
